tests/mlflow/tensorflow/test_tensorflow_model_export.py

```diff
@@ -55,32 +77,108 @@ def test_log_saved_model(self):
+                np.testing.assert_array_equal(saved, results)
+            finally:
+                # Restoring the old logging location.
+                tracking.end_run()
+                tracking.set_tracking_uri(old_tracking_dir)
+
+
+    def test_cat_cols(self):
+        """
+        This tests logging capabilities on datasets with categorical columns.
+        See https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/get_started/regression/imports85.py
+        for reference code.
+        """
+        with TempDir(chdr=False, remove_on_exit=True) as tmp:
+            # Downloading the data into a pandas DataFrame.
+            URL = "https://archive.ics.uci.edu/ml/machine-learning-databases/autos/imports-85.data"
+            path = tf.contrib.keras.utils.get_file(URL.split("/")[-1], URL)
+            # Order is important for the csv-readers, so we use an OrderedDict here.
+            defaults = collections.OrderedDict([
+                ("body-style", [""]),
+                ("curb-weight", [0.0]),
+                ("highway-mpg", [0.0]),
+                ("price", [0.0])
+            ])
+
+            types = collections.OrderedDict((key, type(value[0]))
+                                            for key, value in defaults.items())
+            df = pandas.read_csv(path, names=types.keys(), dtype=types, na_values="?")
+            df = df.dropna()
+
+            # Extract the label from the features dataframe.
+            y_train = df.pop("price")
+
+            # Creating the input training function required.
+            trainingFeatures = {}
+
+            for i in df:
+                trainingFeatures[i] = df[i].values
+
+            input_train = tf.estimator.inputs.numpy_input_fn(trainingFeatures, 
+                                                            y_train.values, 
+                                                            shuffle=False, 
+                                                            batch_size=1)
+
+            # Creating the feature columns required for the DNNRegressor.
+            body_style_vocab = ["hardtop", "wagon", "sedan", "hatchback", "convertible"]
+            body_style = tf.feature_column.categorical_column_with_vocabulary_list(
+                key="body-style", vocabulary_list=body_style_vocab)
+            feature_columns = [
+            tf.feature_column.numeric_column(key="curb-weight"),
+            tf.feature_column.numeric_column(key="highway-mpg"),
+            # Since this is a DNN model, convert categorical columns from sparse
+            # to dense.
+            # Wrap them in an `indicator_column` to create a
+            # one-hot vector from the input.
+            tf.feature_column.indicator_column(body_style),]
+
+            # Build a DNNRegressor, with 2x20-unit hidden layers, with the feature columns
+            # defined above as input.
+            model = tf.estimator.DNNRegressor(
```

Comment:
Sorry tiny nit - could you rename `model` here to `estimator` & `model_predict` -> `estimator_preds` as well?